### PR TITLE
HOSTEDCP-809: Clone CA key/cert to TLS key/cert

### PR DESCRIPTION
**Pull request by bryan-cox**
**What this PR does / why we need it**:
Clone CA key/cert to TLS key/cert as some certificate managers expect to use tls.key/tls.crt rather than ca.key/ca.crt.

**Which issue(s) this PR fixes**:
[HOSTEDCP-809](https://issues.redhat.com/browse/HOSTEDCP-809)

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.

### CODE REVIEW

**Comment by csrwng** [comment]
@bryan-cox this is working for me with the latest passing 4.13.0-ci image. 
Now CI is failing because we've branched for 4.13 and we're saying we don't support 4.14. Will submit a PR to fix that.

File: support/certs/tls.go [hunk 1]
```diff
@@ -30,9 +30,11 @@ const (
 	ValidityOneYear  = 365 * ValidityOneDay
 	ValidityTenYears = 10 * ValidityOneYear
 
-	CAHashAnnotation   = "hypershiftlite.openshift.io/ca-hash"
-	CASignerCertMapKey = "ca.crt"
-	CASignerKeyMapKey  = "ca.key"
+	CAHashAnnotation    = "hypershiftlite.openshift.io/ca-hash"
```
Comment (enxebre) [inline support/certs/tls.go:33]: what's is hypershiftlite?

Comment (bryan-cox) [reply support/certs/tls.go:33]: Hmm not sure. It was brought in with https://github.com/openshift/hypershift/pull/1657.

File: support/certs/tls.go [hunk 2]
```diff
@@ -376,16 +367,20 @@ func ReconcileSignedCert(
...
-	CASignerCertMapKey string
-	CASignerKeyMapKey  string
+	CASignerCertMapKey  string
+	CASignerKeyMapKey   string
+	TLSSignerCertMapKey string
+	TLSSignerKeyMapKey  string
 }
 
 func (s *CAOpts) withDefaults() *CAOpts {
 	if s.CASignerCertMapKey == "" {
```
Comment (enxebre) [inline support/certs/tls.go:377]: why does CASignerCertMapKey dictates both CASignerCertMapKey / TLSSignerKeyMapKey?

File: support/certs/tls.go [hunk 3]
```diff
@@ -30,9 +30,11 @@ const (
 	ValidityOneYear  = 365 * ValidityOneDay
 	ValidityTenYears = 10 * ValidityOneYear
 
-	CAHashAnnotation   = "hypershiftlite.openshift.io/ca-hash"
-	CASignerCertMapKey = "ca.crt"
-	CASignerKeyMapKey  = "ca.key"
+	CAHashAnnotation    = "hypershiftlite.openshift.io/ca-hash"
+	CASignerCertMapKey  = "ca.crt"
+	CASignerKeyMapKey   = "ca.key"
+	TLSSignerCertMapKey = "tls.crt"
```
Comment (enxebre) [inline support/certs/tls.go:36]: can we add a comment here documenting why/how CASignerCertMapKey and CASignerKeyMapKey are different from TLSSignerCertMapKey and TLSSignerKeyMapKey?
And clarifying that the latter is programatically enforced to have the same content than the former?

File: support/certs/tls.go (hunk 2 above)
Comment (bryan-cox) [reply support/certs/tls.go:377]: TLSSignerCertMapKey is setup to mirror CASignerCertMapKey, and TLSSignerKeyMapKey is set to mirror CASignerKeyMapKey.

Originally, I was just change the const values for CASignerCertMapKey, CASignerKeyMapKey; however @csrwng said that would probably be too disruptive, especially if there are existing clusters that need to be upgraded (as in IBM's case). Basically we need to change the CA reconciliation to write the same data to the keys asked for in the Jira ticket.

File: support/certs/tls.go (hunk 3 above)
Comment (bryan-cox) [reply support/certs/tls.go:40]: Added

File: support/certs/tls.go (hunk 2 above)
Comment (bryan-cox) [reply support/certs/tls.go]: Oh I see. Yeah that makes sense. It's been updated.

File: support/certs/tls.go (hunk 1 above)
Comment (csrwng) [reply support/certs/tls.go]: The original intent for that annotation was to cause re-generation of the secret if the CA that original signed it had changed. However, looking at the code, that is not currently used. I think it is valuable though. Looks like we don't currently take into consideration the CA when deciding to sign a new cert or not.

Also, have no idea how the hypershiftlite prefix came to this (This is something I worked on as a demo a while back -> https://github.com/openshift-hive/hypershift-lite) 

I think we should address it separately and not in this PR (decide whether to drop or use but with a proper name)

**Pull request merged**